### DISCUSSION TITLE
resetZoom() with multiple charts

### DISCUSSION THREAD

**Discussion by life-of-e** [chartjs/chartjs-plugin-zoom#583] [Q&A]
I'm having trouble with resetting zoom on a page with multiple charts.  I have two charts created using new Chart() calls so they're each getting a unique ID. Because the charts are related I just wanted to have one button on the page that resets the zoom for both charts.

Clicking the button calls the two charts in order:
window.chart1.resetZoom();
window.chart2.resetZoom();

If I zoom in on the data in chart1 and leave chart2 normal then click the reset button, chart1 resets properly.  But if I zoom in on chart2 leaving chart1 alone and click the button, chart1 zooms in to the same scale as chart2 and neither chart resets.

I'm not sure how to get around this problem because it's not obvious what charts are zoomed in or not.

**Comment by life-of-e** [comment]
So for further details I've got zoom plugin version 1.0.1 and it seems that the plugin is being shared across charts rather than being an instance for each chart. How do I ensure that each chart gets its own copy of the plugin so it's not sharing variables?

**Reply by kurkle** [reply]
The plugin is storing its [state per chart](https://github.com/chartjs/chartjs-plugin-zoom/blob/master/src/state.js), so it should not share variables between charts.
If you assign variables to the plugin, those would be shared.

If the problems you are having originate from the plugin, you should first try upgrading to latest version (1.1.1) and if that does not help, then create a as simple as possible reproduction of the issue.

**Reply by life-of-e** [reply]
Ok, I currently use zoom as an included script so I'm trying to find a CDN where I can download the completed js file for 1.1.1.  As soon as I can find one I'll update and let you know.

**Reply by kurkle** [reply]
https://www.jsdelivr.com/package/npm/chartjs-plugin-zoom

**Reply by life-of-e** [reply]
Great thanks, I've just installed it and it's still happening so I'm testing a trivial example locally first. Where would you like the example?

**Reply by kurkle** [reply]
codepen, jsfiddle or something similar would be good
